src/drive/mobile/containers/FirstUploadModal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react'
+import Modal, { ModalContent } from 'cozy-ui/react/Modal'
+import { connect } from 'react-redux'
+import withPersistentState from '../lib/withPersistentState'
+import styles from '../styles/uploadprogression'
+import { isIos } from '../lib/device'
+
+const MINIMUM_LONG_UPLOAD_FILES_COUNT = 50
+
+class FirstUploadModal extends Component {
+  render() {
+    const { onClose } = this.props
+    const { t } = this.context
+
+    return (
+      <Modal title={t('mobile.first_sync.title')} withCross={false}>
+        <ModalContent>
+          <h4>{t('mobile.first_sync.tips')}</h4>
+          <ul>
+            <li>{t('mobile.first_sync.tip_wifi')}</li>
+            <li>{t('mobile.first_sync.tip_bed')}</li>
+            {isIos() && <li>{t('mobile.first_sync.tip_lock')}</li>}
+          </ul>
+          <p>{t('mobile.first_sync.result')}</p>
+          <button onClick={onClose} className={styles['btn--full-width']}>
+            {t('mobile.first_sync.button')}
+          </button>
+        </ModalContent>
+      </Modal>
+    )
+  }
+}
+
+const showOnlyOnce = WrappedComponent => {
+  return class extends Component {
+    state = {
+      shown: false
+    }
+
+    markAsShown = () => {
+      this.setState({ shown: true })
+    }
+
+    render() {
+      const { uploading } = this.props
+      const { shown } = this.state
+
+      if (uploading && !shown)
+        return <WrappedComponent {...this.props} onClose={this.markAsShown} />
+      else return null
+    }
+  }
+}
+
+const OnlyOnceFirstUploadModal = withPersistentState(
+  showOnlyOnce(FirstUploadModal),
```

Comment:
I cannot understand the `showOnlyOnce` HOC as it uses an internal state controlled by a `onClose` button.
I guess the wanted behavior is to use `withPersistentState` to persist the internal state to actually show only once the `FirstUploadModal`. So the `showOnlyOnce` HOC takes its name because we use it with `withPersistentState`.

Reply:
This is all correct, so I don't understand what problem you are pointing out here. Can you clarify please?

Reply:
`showOnlyOnce` utilisé tout seul ne fait pas vraiment ce qu'il prétend faire:

```jsx
const Hello = ({ name, onClose }) => (
  <div>
    <div>Hello {name}</div>
    <button onClick={onClose}>close</button>
  </div>
)
const ShowOnlyOnceHello = showOnlyOnce(Hello)
reactDOM.render(<ShowOnlyOnceHello />, document.getElementById('root')
```

c'est plutôt un truc qui dit qu'on peut le fermer.

C'est avec `withPersistentState` (un HOC un peu bizarre puisqu'il casse un peu le lifecycle des composants (p)react) qu'on peut dire qu'il montre seulement une fois.

Reply:
Je l'ai renommé, mais c'est pas beaucoup mieux — et je n'ai pas de meilleure idée.